Podverse/PVAuth.swift

```diff
@@ -7,222 +7,104 @@
-    }
-    
-    func showAuth0LockLoginVC (vc: UIViewController) {
-        let lock = A0Lock.shared()
-        let controller = lock.newLockViewController()
-        controller?.closable = true
-        
-        controller?.onAuthenticationBlock = {(profile, token) in
-            guard let idToken = token?.idToken else {
-                return
+    static let shared = PVAuth()
+
+    func showAuth0Lock (vc: UIViewController, completion:(() -> ())?) {
+
+        Lock
+            .classic()
+            .withOptions{
+                $0.closable = true
+                $0.oidcConformant = false
+                $0.customSignupFields = [
+                    CustomTextField(name: "Name", placeholder: "your name (optional)")
+                ]
+                $0.usernameStyle = [.Email]
             }
-            
-            guard let userId = profile?.userId else {
-                return
+            .withStyle {
+                $0.title = "Podverse Login"
+                $0.primaryColor = UIColor(red: 0.15, green: 0.41, blue: 0.70, alpha: 1.0)
+                $0.hideTitle = true
             }
-            
-            self.updateOwnedItemsThenSwitchToNewUser(idToken: idToken, userId: userId, completionBlock: { () in
-                controller?.dismiss(animated: true, completion: nil)
-            })
-            
-        }
-        
-        controller?.onUserDismissBlock = {() in
-            self.loginAsAnon()
-            self.delegate?.authFinished()
-        }
-        
-        lock.present(controller, from: vc, presentationStyle: .custom)
-    }
-    
-    func showAuth0LockSignUpVC (vc: UIViewController) {
-        let lock = A0Lock.shared()
-        let controller = lock.newSignUpViewController()
-        
-        controller?.onAuthenticationBlock = {(profile, token) in
-            guard let idToken = token?.idToken else {
-                return
+            .onAuth {
+                
+                DispatchQueue.main.async {
+                    vc.dismiss(animated: true, completion: nil)
+                }
+                
+                guard let accessToken = $0.accessToken, let idToken = $0.idToken else {
+                    return
+                }
+                
+                Auth0
+                    .authentication()
+                    .userInfo(withAccessToken: accessToken)
+                    .start { result in
+                        switch result {
+                        case .success(let profile):
+                            self.setUserInfo(idToken: idToken, userId: profile.sub, userName: profile.nickname)
+                        case .failure(let error):
+                            self.setUserInfo(idToken: idToken, userId: nil, userName: nil)
+                            print(error)
+                        }
+                        
+                        // TODO: uh is there a better way to do this completion?
+                        if let completion = completion {
```

Comment:
To the question above: Yes there is. Just call `completion?()`. It is an optional and does not need unwrapping. If it is nil, its will simply do nothing